src/borg/archiver.py

```diff
@@ -1199,14 +1203,16 @@ def _delete_archives(self, args, repository):
         with Cache(repository, key, manifest, progress=args.progress, lock_wait=self.lock_wait) as cache:
             msg_delete = 'Would delete archive: {} ({}/{})' if dry_run else 'Deleting archive: {} ({}/{})'
             msg_not_found = 'Archive {} not found ({}/{}).'
-            _logger = logging.getLogger('borg.output.list') if self.output_list else logger
+            logger_list = logging.getLogger('borg.output.list')
             for i, archive_name in enumerate(archive_names, 1):
                 try:
                     archive_info = manifest.archives[archive_name]
                 except KeyError:
                     logger.warning(msg_not_found.format(archive_name, i, len(archive_names)))
                 else:
-                    _logger.info(msg_delete.format(format_archive(archive_info), i, len(archive_names)))
+                    logger.info(msg_delete.format(format_archive(archive_info), i, len(archive_names)))
+                if self.output_list:
+                    logger_list.info(msg_delete.format(format_archive(archive_info), i, len(archive_names)))
```

Comment:
you need to remove the 1st log call and indent the 2nd call (so it is within else block).